src/formatting.rs

```diff
@@ -683,6 +701,40 @@ fn parse_crate(
     Err(ErrorKind::ParseError)
 }
 
+struct SilentOnIgnoredFilesEmitter {
+    ignore_path_set: Rc<IgnorePathSet>,
+    source_map: Rc<SourceMap>,
+    emitter: EmitterWriter,
+    has_non_ignorable_parser_errors: bool,
+    can_reset: Rc<RefCell<bool>>,
+}
+
+impl Emitter for SilentOnIgnoredFilesEmitter {
+    fn emit_diagnostic(&mut self, db: &DiagnosticBuilder<'_>) {
+        if let Some(primary_span) = &db.span.primary_span() {
+            let file_name = self.source_map.span_to_filename(*primary_span);
+            match file_name {
+                syntax_pos::FileName::Real(ref path) => {
+                    if self
+                        .ignore_path_set
+                        .is_match(&FileName::Real(path.to_path_buf()))
+                    {
+                        if !self.has_non_ignorable_parser_errors {
+                            *self.can_reset.borrow_mut() = true;
```

Comment:
I first thought that we could simply call `db.handler.reset_err_count();` here instead, but looks like the rustc parser bumps the error count **after** calling `emit_diagnostic`:

https://github.com/rust-lang/rust/blob/f2023ac599c38a59f86552089e6791c5a73412d3/src/librustc_errors/lib.rs#L781-L791

Also like you mentioned, resetting the error count may cause skipping important parsing errors.

Reply:
> I first thought that we could simply call db.handler.reset_err_count();

That's what I was expecting originally too 👍 

> here instead, but looks like the rustc parser bumps the error count after calling emit_diagnostic:

Thank you so much for finding that! I was baffled how calling `reset_err_count()` inside `emit_diagnostic` seemed to be doing nothing, but that makes much more sense now